### PR TITLE
Major/uppsf 649 v2 upp logging

### PR DESCRIPTION
**Pull request by epavlova**
v2 of UPP logging library

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,34 +1,52 @@
 # go-logger
-Logging library in GO, used for structural logging inside UPP (mostly required for monitoring publish events). It is based on the [Logrus](https://github.com/sirupsen/logrus) implementation.
+
+Logging library in Go, used for structural logging inside UPP. It is wrapper of [logrus](https://github.com/sirupsen/logrus) library for structrural logging in Go.
```
Comment (MiroslavGatsanoga) [inline README.md]: I think the correct term is "structured logging".

File: README.md [hunk 2]
```diff
@@ -71,3 +89,11 @@ func TestSomething(t *testing.T) {
 }
 
 ```
+
+### Installation
```
Comment (MiroslavGatsanoga) [inline README.md]: Do we need installation instructions for a library?

File: entry.go [hunk 3]
```diff
@@ -1,35 +1,77 @@
...
 import (
-	"github.com/sirupsen/logrus"
 	"strconv"
 	"time"
+
+	"github.com/sirupsen/logrus"
 )
 
-type logEntry struct {
+// logEntry is wrapper around logrus.Entry with a few methods adding UPP specific keys to the log entries.
```
Comment (MiroslavGatsanoga) [inline entry.go]: The doc comment should match the name of the struct i.e. logEntry should become LogEntry.

File: entry_test.go [hunk 4]
```diff
@@ -1,77 +1,118 @@
...
 	assert.Equal(t, "a info message", hook.LastEntry().Message)
-	isValid, err := strconv.ParseBool(hook.LastEntry().Data["isValid"].(string))
+	isValid, err := strconv.ParseBool(hook.LastEntry().Data[DefaultKeyIsValid].(string))
 	assert.NoError(t, err)
 	assert.False(t, isValid)
-	assert.Equal(t, "tid_test", hook.LastEntry().Data["transaction_id"])
+	assert.Equal(t, "tid_test", hook.LastEntry().Data[DefaultKeyTransactionID])
+}
+
+func TestLogEntry_WithMonitoringEvent(t *testing.T) {
```
Comment (MiroslavGatsanoga) [inline entry_test.go]: The underscore in the test name seems inconsistent with the rest of the tests.

File: entry_test.go [hunk 5]
```diff
@@ -1,77 +1,118 @@
...
+	assert.Len(t, hook.Entries, 1)
+	assert.Len(t, hook.LastEntry().Data, 5)
+	assert.Equal(t, logrus.InfoLevel, hook.LastEntry().Level)
+	assert.Equal(t, "an-event", hook.LastEntry().Data[conf.KeyEventName])
+	assert.Equal(t, "true", hook.LastEntry().Data[DefaultKeyMonitoringEvent])
+	assert.Equal(t, "some-content", hook.LastEntry().Data[DefaultKeyContentType])
+	assert.Equal(t, "tid_test", hook.LastEntry().Data[DefaultKeyTransactionID])
+}
+
+func TestLogEntry_WithCategorisedEvent(t *testing.T) {
```
Comment (MiroslavGatsanoga) [inline entry_test.go]: The underscore in the test name seems inconsistent with the rest of the tests.

File: exported.go [hunk 6]
```diff
@@ -22,139 +20,55 @@ func WithField(key string, value interface{}) LogEntry {
...
+			ulog.keyConf.KeyEventName:       eventName,
+			ulog.keyConf.KeyContentType:     contentType,
+		})
+	return e.WithTransactionID(tid)
 }
 
-// Fatalln logs a message at level Fatal on the standard logger.
-func Fatalln(args ...interface{}) {
-	log.Fatalln(args...)
+// WithMonitoringEvent creates an entry from the standard logger and adds categorised event fields to it.
```
Comment (MiroslavGatsanoga) [inline exported.go]: The doc comment is not correct.

**Comment by epavlova** [comment]
The used time format is RFC3339Nano - looks like "2019-09-12T11:41:12.186416+03:00".
It's open question whether we prefer RFC3339 - "2019-09-12T11:41:12+03:00".

File: logkeys.go [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+		KeyEventName:       DefaultKeyEventName,
+		KeyMonitoringEvent: DefaultKeyMonitoringEvent,
+		KeyContentType:     DefaultKeyContentType,
+		KeyEventCategory:   DefaultKeyEventCategory,
+		KeyEventMsg:        DefaultKeyEventMsg,
+	}
+}
+
+// GetFullKeyNameConfig returns KeyNamesConfig that has all key names from the input conf and
+// if there are key names missing from the input conf, the default key names are used
```
Comment (MiroslavGatsanoga) [inline logkeys.go]: Missing full stop at the end of the doc comment.

File: logkeys.go [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+func GetFullKeyNameConfig(conf KeyNamesConfig) *KeyNamesConfig {
+	defaultConfig := GetDefaultKeyNamesConfig()
+	resConfig := conf
```
Comment (MiroslavGatsanoga) [inline logkeys.go]: We don't need `resConfig` here, because `conf` is already a copy (it is a struct passed by value).

File: logkeys.go [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+
+	if keyLogLevel := conf.KeyLogLevel; keyLogLevel == "" {
```
Comment (MiroslavGatsanoga) [inline logkeys.go]: All conditional statements could be made more readable if they are in the form:
```
if conf.KeyLogLevel == "" {
...
```

**Review by mitagg — APPROVED** [review]

**Review by rnov — APPROVED** [review]

**Review by gotha — APPROVED** [review]

**Review by MiroslavGatsanoga — APPROVED** [review]

**Pull request merged**